RULES.md

```diff
@@ -292,6 +302,24 @@ It is not allowed to compute any kind of pairwise metrics between the public wor
 - On a held-out workload, submissions are not allowed to find the nearest neighbor among the public workloads to set any hyperparameter.
 
 </details>
+<br>
+
+Valid submissions must rely on new algorithmic or mathematical ideas and should not use software engineering approaches to speed up basic operations in `PyTorch`, `JAX`, their dependencies, the operating system, or the hardware.
+
+<details>
+<summary>Examples:</summary>
+
```

Comment:
Add somewhere:
"In general, submissions can make clever, judicious, and efficient use of public APIs in JAX and/or PyTorch but should not be trying to optimize the internals of primitive operations and standard dependencies."